slither/detectors/variables/var_read_using_this.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+from slither.core.cfg.node import Node
+from slither.core.declarations import Function, SolidityVariable
+from slither.detectors.abstract_detector import AbstractDetector, DetectorClassification
+from slither.slithir.operations.high_level_call import HighLevelCall
+
+
+class VarReadUsingThis(AbstractDetector):
+    ARGUMENT = "var-read-using-this"
+    HELP = "Contract reads its own variable using `this`"
+    IMPACT = DetectorClassification.OPTIMIZATION
+    CONFIDENCE = DetectorClassification.MEDIUM
```

Comment:
What is the reason for the medium confidence?

Reply:
I didn't give it much thought. I suppose it can be high as I don't think it will be noisy